src/Database/Statement/Statement.php

```diff
@@ -297,4 +285,24 @@ public function queryString(): string
     {
         return $this->statement->queryString;
     }
+
+    /**
+     * Get the inner iterator
+     *
+     * @return \Generator
+     */
+    public function getIterator(): Generator
+    {
+        $this->statement->setFetchMode(PDO::FETCH_ASSOC);
+
+        foreach ($this->statement as $row) {
+            foreach ($this->resultDecorators as $decorator) {
+                $row = $decorator($row);
+            }
+
+            yield $row;
+        }
+
+        $this->closeCursor();
+    }
```

Comment:
I'm not sure I see how this and fetchAll() should co-exist. Won't the statement be empty if we call one or the other?